File: frontend/src/types/apiKey.ts
export interface ApiKey {
  id: string;
  agent_id: string;
  name: string;
  key_prefix: string;
  created_at: string;
  last_used_at: string | null;
}

export interface ApiKeyCreateRequest {
  name: string;
}

export interface ApiKeyCreateResponse extends ApiKey {
  key: string;
}

File: backend/app/schemas/api_key.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ApiKeyCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)


class ApiKeyResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    agent_id: uuid.UUID
    name: str
    key_prefix: str
    created_at: datetime
    last_used_at: datetime | None


class ApiKeyCreateResponse(ApiKeyResponse):
    """Returned only once at creation time — includes the raw key."""
    key: str